lua/nvim-treesitter/query_predicates.lua

```diff
@@ -75,6 +75,42 @@ query.add_predicate("has-ancestor?", has_ancestor)
 
 query.add_predicate("has-parent?", has_ancestor)
 
+---@param match (TSNode|nil)[]
+---@param pattern string
+---@param bufnr integer
+---@param pred string[]
+---@return boolean|nil
+local function has_child(match, pattern, bufnr, pred)
+  if not valid_args(pred[1], pred, 2) then
+    return
+  end
+
+  local node = match[pred[2]]
+  local child_types = { unpack(pred, 3) }
+  if not node then
+    return true
+  end
+
+  local just_direct_child = pred[1]:find("has-child", 1, true)
+
+  for _, child in ipairs(node:named_children()) do
+    if vim.tbl_contains(child_types, child:type()) then
+      return true
+    end
+    if not just_direct_child then
+      local child_has_child = has_child({ [pred[2]] = child }, pattern, bufnr, pred)
+      if child_has_child then
+        return true
+      end
+    end
+  end
+  return false
+end
+
+query.add_predicate("has-child?", has_child)
+
+query.add_predicate("has-descendant?", has_child)
```

Comment:
I'd choose one or the other. `has-descendant` seems like a better fit.

Reply:
done